pkg/providers/instanceprofile/instanceprofile.go

```diff
@@ -62,7 +73,13 @@ func (p *DefaultProvider) Get(ctx context.Context, instanceProfileName string) (
 	return out.InstanceProfile, nil
 }
 
-func (p *DefaultProvider) Create(ctx context.Context, instanceProfileName string, roleName string, tags map[string]string) error {
+func (p *DefaultProvider) Create(ctx context.Context, instanceProfileName string, roleName string, tags map[string]string, nodeClassUID string) error {
+	_, err := p.iamapi.GetRole(ctx, &iam.GetRoleInput{
```

Comment:
We should be caching this call - we could either store it in the existing cache and create unique keys for roles / instance profiles or create another cache. Doesn't matter much either way. It should just be a relatively short TTL to ensure that we react when a customer creates a role after creating the nodeclass.